addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.network.internal.dhcp;
+
+import java.io.IOException;
+import java.net.BindException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+
+import org.openhab.binding.network.internal.dhcp.DHCPPacket.BadPacketException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Receives UDP messages and try to parse them as DHCP messages.
+ * First try to listen to the DHCP port 67 and if that failes because of missing access rights,
+ * port 6767 will be opened instead (and the user is required to setup a port forarding).
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class DHCPPacketListenerServer extends Thread {
+    private byte[] buffer = new byte[1024];
+    DatagramSocket dsocket;
+    private DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+    boolean willbeclosed = false;
+    Logger logger = LoggerFactory.getLogger(DHCPPacketListenerServer.class);
+    private boolean useUnprevilegedPort = false;
+    private final IPRequestReceivedCallback listener;
+
+    DHCPPacketListenerServer(IPRequestReceivedCallback listener) throws SocketException, BindException {
+        this.listener = listener;
+        try {
+            dsocket = new DatagramSocket(null);
```

Comment:
It would be clearer to seperate this body an the body's of the catch with a parameter for the port number.